pymoveit2/moveit2.py

```diff
@@ -48,10 +63,9 @@ def __init__(
         base_link_name: str,
         end_effector_name: str,
         group_name: str = "arm",
-        execute_via_moveit: bool = False,
+        use_move_group_action: bool = False,
         ignore_new_calls_while_executing: bool = False,
         callback_group: Optional[CallbackGroup] = None,
-        follow_joint_trajectory_action_name: str = "joint_trajectory_controller/follow_joint_trajectory",
```

Comment:
Are these parameters the only breaking changes introduced by this PR for the public API?

If so, do you think it would be possible to keep the original parameters while indicating that they are *deprecated* to prevent sudden breaking changes? If it is possible, it would be great!

For example, `follow_joint_trajectory_action_name` could potentially be ignored, and the meaning of `execute_via_moveit` could be combined with the new `use_move_group_action`. If they are set to non-default values, then a deprecation warning would be printed.

Reply:
While these parameters are the only explicit breaking changes to the public API, there is an additional **implicit** change because execution now goes through the `/execute_trajectory` action (of type `moveit_msgs/ExecuteTrajectory`) and not the action specified `follow_joint_trajectory_action_name` (of type `control_msgs/FollowJointTrajectory`). In cases where the user has configured their controller with `moveit2`, that should be fine. But in the off-chance that they have not configured their controller with `moveit2`, then this class will no longer be able to execute their robot.

I considered adding an either-or system where users can either use `ExecuteTrajectory` or use `FollowJointTrajectory`. However, the issue is that cancellation logic is very different for each. Whereas `ExecuteTrajectory` can be canceled via topic publication, `FollowJointTrajectory` requires making a cancellation request to the action server, which can be accepted or rejected. That’s why I ended up removing execution via `FollowJointTrajectory` altogether, in favor of execution via `ExecuteTrajectory`. I think that also makes more sense, because this class is an interface for `moveit2`, so it makes sense for execution to also go through `moveit2`.

Re. your changes about deprecation, I made the changes, but I don’t think the check for non-default values will work well for `follow_joint_trajectory_action_name `. Because some existing users may not pass in a custom value because the default value is the true name for their controller’s action. For those users, and the deprecation warning won’t show, so they won’t know why commands aren’t being sent to the controller.

Reply:
> there is an additional **implicit** change because execution now goes through the `/execute_trajectory` action (of type `moveit_msgs/ExecuteTrajectory`) and not the action specified `follow_joint_trajectory_action_name` (of type `control_msgs/FollowJointTrajectory`). 

> ... cancellation logic is very different for each. Whereas `ExecuteTrajectory` can be canceled via topic publication, `FollowJointTrajectory` requires making a cancellation request to the action server, which can be accepted or rejected. That’s why I ended up removing execution via `FollowJointTrajectory` altogether, in favor of execution via `ExecuteTrajectory`. I think that also makes more sense, because this class is an interface for `moveit2`, so it makes sense for execution to also go through `moveit2`.

I think you are right, and this change should be fine.

> I considered adding an either-or system where users can either use `ExecuteTrajectory` or use `FollowJointTrajectory`. 

As long as `ExecuteTrajectory` covers everything that's required for basic usage with simple arms and grippers, then sticking to this interface should be fine.

> Re. your changes about deprecation, I made the changes, but I don’t think the check for non-default values will work well for `follow_joint_trajectory_action_name `. Because some existing users may not pass in a custom value because the default value is the true name for their controller’s action. For those users, and the deprecation warning won’t show, so they won’t know why commands aren’t being sent to the controller. So fi we keep `follow_joint_trajectory_action_name`, perhaps we should print the deprecation warning every time? But that will be noisy for folks who don't use that parameter...

Thank you. I think it is fine. Alternatively, the default value for `follow_joint_trajectory_action_name` could be set to a special "DEPRECATED" string, an empty string or even None. It does not really matter what the default is if it is not used anyway. :)